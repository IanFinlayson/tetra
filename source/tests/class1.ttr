class Square:
    size as real

    def init(size real):
        self.size = size
        print("In cons, size = ", self.size, "\n")

    def area() real:
        return size ** 2.0

def main():
    s = Square(5.0)
    print("Outside, size = ", s.size, "\n")

