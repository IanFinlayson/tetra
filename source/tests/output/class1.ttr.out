In cons, size = 5
Outside, size = 5

